frontend/src/pages/blog/Article.js

```diff
@@ -4,9 +4,28 @@ import { useArticlesContext } from "../../hooks/useArticlesContext";
 
 const Article = () => {
   let { id } = useParams();
-  const { articles } = useArticlesContext();
-  const article = articles.filter((a) => a._id === id)[0];
-  return <ArticleView article={article} />;
+  const { articles, dispatch } = useArticlesContext();
+
+  const fetchArticle = async () => {
+      const response = await fetch(`/api/articles/${id}`);  // fetch the article (only this one) from the database
+      if (response.ok) {
+        const article = await response.json();
+        dispatch({ type: "ADD", payload: article });       // if it exists, add it to the ArticlesContext
```

Comment:
I think you should do
```suggestion
        dispatch({ type: "SET", payload: [article] }); // if it exists, add it to the ArticlesContext
```
as `SET` is defined (and used in `frontend/src/pages/blog/Blog.js`) it is the proper way to do what you are trying to do!
And it also remove the need of redefining `ADD` as you did :wink: 
@atxr

Reply:
No I don't think this is a good idea.
Actually if your context is not empty it will overwrite it.
Take the following example:
-> You browse an article
-> From this article, you browse another article
-> You finally go back to the first article
In this example, if you `SET`, the first article will be overwritten when browsing the second one. So the app will fetch again the first article in the third step.
I hope it makes sense :)

Reply:
@atxr answered in my stead, if you use `SET` for some reason but there are already articles loaded in, you will overwrite everything - so definetly not a good idea

Reply:
@atxr 
Ok you're right I didn't see things this way, but then, what's the whole point of *having* a `SET`?

Reply:
`SET` is used [here](https://github.com/iScsc/iscsc.fr/blob/main/frontend/src/pages/blog/Blog.js#L15)
It's used when you load the `/` route to fetch all the article.

Reply:
Yes I already saw it but now that `ADD` has been patched what's the difference between using `ADD` when empty and using `SET` (apart from the list), does it not seem confusing to you ?

Reply:
`ADD` adds only one article when `SET` can add multiple at the same time

Reply:
Ok, I already got this point but while thinking about yesterday I finally understand why we need both, my problem was a little pointy but I get it!